src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final int TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromFile = readFromFile(fromFileName);
+        System.out.println(dataFromFile);
+        String report = getReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    public List<String> readFromFile(String fileName) {
+        File file = new File(fileName);
+        List<String> result = new ArrayList<>();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line = reader.readLine();
+            while (line != null) {
+                result.add(line);
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file!", e);
```

Comment:
be informative)
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/5ddd97fd-0ada-4523-b5c9-e6da1faed87b)